src/main/kotlin/app/revanced/patches/youtube/layout/spoofappversion/bytecode/patch/SpoofAppVersionPatch.kt

```diff
@@ -59,6 +59,10 @@ class SpoofAppVersionPatch : BytecodePatch(
             )
         } ?: return SpoofAppVersionFingerprint.toErrorResult()
 
+        SettingsPatch.addString("revanced_user_notice_message_app_version_spoofing",
```

Comment:
Where is this string being used?

Reply:
The way I understood is that the string is used for the dialog string when toggling a setting. In that case, it is weird to inject the string like that. A better API would be that you can pass the string along with the setting it is associated with and let the settings patch add the string. What is the reason for separating it and adding the string manually?

Reply:
Currently, `SettingsPatch` incorporates a lot of boilerplate with how it works. Technically it is possible to reduce boilerplate and improve API design in the following way:


```diff
SettingsPatch.PreferenceScreen.LAYOUT.addPreferences(
    SwitchPreference(
        "revanced_spoof_app_version",
-        StringResource("revanced_spoof_app_version_title", "Spoof app version"),
+        title = "Spoof app version",
-        false,
+        default = false,
-        StringResource("revanced_spoof_app_version_summary_on", "Version spoofed to 17.30.34"),
+        summaryOn = "Version spoofed to 17.30.34",
-        StringResource("revanced_spoof_app_version_summary_off", "Version not spoofed")
+        summaryOff = "Version not spoofed"
   )
)
````

This way the API can be used conveniently:

```kotlin
SettingsPatch.PreferenceScreen.LAYOUT.addPreferences(
    SwitchPreference(
        "revanced_spoof_app_version",
        "Spoof app version"
        false,
        "Version spoofed to 17.30.34",
        "Version not spoofed",
        dialogString = "App version will be spoofed to 17.30.34. This will give the old UI layout, but unknown side effects may occur. If later turned off, the old UI layout may remain until you log out or clear the app data."
   )
)
```

The patch would simply autogenerate the proper keys based on the property, reducing boilerplate. For that reason, for now, adding the string should be incorporated into the API the following way. An example:

```diff
SettingsPatch.PreferenceScreen.LAYOUT.addPreferences(
    SwitchPreference(
        "revanced_spoof_app_version",
        StringResource("revanced_spoof_app_version_title", "Spoof app version"),
        false,
        StringResource("revanced_spoof_app_version_summary_on", "Version spoofed to 17.30.34"),
        StringResource("revanced_spoof_app_version_summary_off", "Version not spoofed")
+       dialogString = StringResource("revanced_user_notice_message_app_version_spoofing", "App version will be spoofed to 17.30.34. This will give the old UI layout, but unknown side effects may occur. If later turned off, the old UI layout may remain until you log out or clear the app data.")
    )
)
````

Reply:
I was having something more simpler in mind. You could simply make `SettingsPatch` add the string and use the key for the string in `SettingsEnum`. This way you don't have to manually use `SettingsPatch.addString()`.